src/main/scala/com/microsoft/hyperspace/index/IndexConfig.scala

```diff
@@ -28,25 +28,38 @@ import java.util.Locale
 case class IndexConfig(
     indexName: String,
     indexedColumns: Seq[String],
-    includedColumns: Seq[String] = Seq()) {
+    includedColumns: IncludedColumnsConfig = IncludedColumnsConfig()) {
   if (indexName.isEmpty || indexedColumns.isEmpty) {
     throw new IllegalArgumentException("Empty index name or indexed columns are not allowed.")
   }
 
   val lowerCaseIndexedColumns = toLowerCase(indexedColumns)
-  val lowerCaseIncludedColumns = toLowerCase(includedColumns)
-  val lowerCaseIncludedColumnsSet = lowerCaseIncludedColumns.toSet
+  val lowerCaseIncludedColumnsIncludeSet = includedColumns.lowerCaseIncludeColumnsSet
+  val lowerCaseIncludedColumnsExcludeSet = includedColumns.lowerCaseExcludeColumnsSet
 
   if (lowerCaseIndexedColumns.toSet.size < lowerCaseIndexedColumns.size) {
     throw new IllegalArgumentException("Duplicate indexed column names are not allowed.")
   }
 
-  if (lowerCaseIncludedColumnsSet.size < lowerCaseIncludedColumns.size) {
-    throw new IllegalArgumentException("Duplicate included column names are not allowed.")
+  if (lowerCaseIncludedColumnsIncludeSet.size < includedColumns.lowerCaseIncludeColumns.size ||
+      lowerCaseIncludedColumnsExcludeSet.size < includedColumns.lowerCaseExcludeColumns.size) {
+    throw new IllegalArgumentException(
+      "Duplicate include or exclude column names in included columns are not allowed.")
+  }
+
+  if (lowerCaseIncludedColumnsIncludeSet.nonEmpty && lowerCaseIncludedColumnsExcludeSet.nonEmpty) {
+    throw new IllegalArgumentException(
+      "IncludedColumnsConfig include and exclude columns can not both be non-empty.")
+  }
+
+  if (lowerCaseIncludedColumnsIncludeSet.intersect(lowerCaseIncludedColumnsExcludeSet).nonEmpty) {
```

Comment:
**To reviewers:** This check is actually redundant and should be dropped, as both lists can not be non-empty and that is verified in the previous check. I will drop this one.